docs/security/faq.md

```diff
@@ -0,0 +1,188 @@
+To prevent duplicates, a very reliable script was written that performs all relevant operations atomically.
+Thus RTL indexes can not be duplicated even in case of power failure at any moment (assuming reliable `mv` and `fsync`).
+
+## Which security techniques does this project use?
+
+* Isolation
+* Principle of least privilege
+* Automation that prevents security-related mistakes
+* Declarative programming with secure defaults (security is always opt-out, never opt-in)
+* Code signing and verification
+* Build in isolated ennvironment
+* Determministic builds
+
+## Which security features are rare or unique to this project?
+
+**Important: the following is based on possibly limited information. If you know of a similar project with a security feature
+mentioned here that's not attributed, please report.**
+
+* This is the only project with a clear policy on forks.
+* To my knowledge this project is the only one that uses `btc-rpc-proxy` to protect `bitcoind` out-of-the-box.
+* This project was the first one to implement any kind of RPC filtering for `bitcoind`, long before it was available in `bitcoind` itself.
+  This was later implemented in `nix-bitcoin`, which is the only other project that does this.
+* This project was the first to implement a special macaroon that doesn't allow BTCPayServer to steal sats from LND.
+  This was later copied by `nix-bitcoin` and it's the only other project that does this.
+  Note that this is meant as a compliment to `nix-bitcoin` - copying important security features is a very good thing!
+* This project implements SSO for several applications.
+  The only other known project that does this is BTCPayServer-docker.
+  The application handling authentication is intentionally much, much simpler than BTCPayServer, thus the attack surface is lower.
+* This project sandboxes cookeis of different apps to avoid them being passed between each-other.
+  There's no other known project that does this.
+* The keys of foreign developers are verified using memory-safe GPG implementation [`sqck`](https://github.com/Kixunil/sqck).
+  No other known project does this.
+* All official builds are performed in a dedicated Qubes VM.
+  No other known project does this.
+* Perhaps except `nix-bitcoin`, no other project attempts deterministic builds.
+  (Only ThunderHub is undeterministic here.)
+* The only other known project that uses declarative approach is `nix-bitcoin`.
+
+## Is it good idea to put my lifesavings into this node?
+
+No.
+
+## Why doesn't this project use Docker?
+
+Docker was **not** designed as a security mechanism and shouldn't be used as such.
+There were actually numerous cases of escaping from Docker containers.
+The Docker daemon adds its own attack surface.
+
+That being said, Docker is just a collection of several mechanisms, some of which may be beneficial for security.
+This project already uses some of them and maybe it will use more in the future (needs to be weighted against disadvantages).
```

Comment:
can you name an example? just for curiosity, might lead to docs clutter and future maintenance burden.

Reply:
Private mounts (as set in `systemd` - `PrivateTmp`). Also Docker actually is using system users too, it just adds a layer of indirection that shouldn't have effect on security. E.g. containers can pretend that application runs under UID 0 (root) without actually running as root.

This has no security benefit apart from being able to run an application that the author was too lazy to make available for non-root users. I wouldn't touch such app with ten feet pole and there's no such app in the repository.